hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java

```diff
@@ -0,0 +1,180 @@
+        cacheConfig.setCacheLoaderFactory(FactoryBuilder.factoryOf(TestCacheLoader.class));
+        return cacheConfig;
+    }
+
+    @Override
+    protected HazelcastInstance getHazelcastInstance() {
+        return hazelcastInstance;
+    }
+
+    private static String getValueOfKey(String key) {
+        return "value_of_" + key;
+    }
+
+    private Map<String, String> createAndFillEntries() {
+        final int ENTRY_COUNT_PER_PARTITION = 3;
+        Node node = getNode(hazelcastInstance);
+        int partitionCount = node.getPartitionService().getPartitionCount();
+        Map<String, String> entries = new HashMap<String, String>(partitionCount * ENTRY_COUNT_PER_PARTITION);
+
+        for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+            for (int i = 0; i < ENTRY_COUNT_PER_PARTITION; i++) {
+                String key = generateKeyForPartition(hazelcastInstance, partitionId);
+                String value = getValueOfKey(key);
+                entries.put(key, value);
+            }
+        }
+
+        return entries;
+    }
+
+    @Test
+    public void testLoadAll() throws InterruptedException {
+        ICache<String, String> cache = createCache();
+        String cacheName = cache.getName();
+        Map<String, String> entries = createAndFillEntries();
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        cache.loadAll(entries.keySet(), true, new CompletionListener() {
+            @Override
+            public void onCompletion() {
+                latch.countDown();
+            }
+
+            @Override
+            public void onException(Exception e) {
+                latch.countDown();
+            }
+        });
+
+        latch.await(60, TimeUnit.SECONDS);
+
+        // Verify that load-all works
+        for (Map.Entry<String, String> entry : entries.entrySet()) {
+            String key = entry.getKey();
+            String expectedValue = entries.get(key);
+            String actualValue = cache.get(key);
+            assertEquals(expectedValue, actualValue);
+        }
+
+        Node node = getNode(hazelcastInstance);
+        InternalPartitionService partitionService = node.getPartitionService();
+        SerializationService serializationService = node.getSerializationService();
+
+        // Verify that backup of load-all works
+        for (Map.Entry<String, String> entry : entries.entrySet()) {
+            String key = entry.getKey();
+            String expectedValue = entries.get(key);
+            Data keyData = serializationService.toData(key);
+            int keyPartitionId = partitionService.getPartitionId(keyData);
+            for (int i = 0; i < INSTANCE_COUNT; i++) {
+                Node n = getNode(hazelcastInstances[i]);
+                ICacheService cacheService = n.getNodeEngine().getService(ICacheService.SERVICE_NAME);
+                ICacheRecordStore recordStore = cacheService.getRecordStore("/hz/" + cacheName, keyPartitionId);
```

Comment:
right, we get them from `ICacheRecordStore` itself directly without any operation.